plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/element/ImplementVspCommand.java

```diff
@@ -0,0 +1,295 @@
+            return this;
+        }
+
+        public Builder networkDomainUuid(String networkDomainUuid) {
+            this._networkDomainUuid = networkDomainUuid;
+            return this;
+        }
+
+        public Builder networkUuid(String networkUuid) {
+            this._networkUuid = networkUuid;
+            return this;
+        }
+
+        public Builder networkName(String networkName) {
+            this._networkName = networkName;
+            return this;
+        }
+
+        public Builder vpcOrSubnetUuid(String vpcOrSubnetUuid) {
+            this._vpcOrSubnetUuid = vpcOrSubnetUuid;
+            return this;
+        }
+
+        public Builder isL2Network(boolean isL2Network) {
+            this._isL2Network = isL2Network;
+            return this;
+        }
+
+        public Builder isL3Network(boolean isL3Network) {
+            this._isL3Network = isL3Network;
+            return this;
+        }
+
+        public Builder isVpc(boolean isVpc) {
+            this._isVpc = isVpc;
+            return this;
+        }
+
+        public Builder isShared(boolean isShared) {
+            this._isShared = isShared;
+            return this;
+        }
+
+        public Builder domainTemplateName(String domainTemplateName) {
+            this._domainTemplateName = domainTemplateName;
+            return this;
+        }
+
+        public Builder isFirewallServiceSupported(boolean isFirewallServiceSupported) {
+            this._isFirewallServiceSupported = isFirewallServiceSupported;
+            return this;
+        }
+
+        public Builder dnsServers(List<String> dnsServers) {
+            this._dnsServers = dnsServers;
+            return this;
+        }
+
+        public Builder ingressFirewallRules(List<Map<String, Object>> ingressFirewallRules) {
+            this._ingressFirewallRules = ingressFirewallRules;
+            return this;
+        }
+
+        public Builder egressFirewallRules(List<Map<String, Object>> egressFirewallRules) {
+            this._egressFirewallRules = egressFirewallRules;
+            return this;
+        }
+
+        public Builder acsFipUuid(List<String> acsFipUuid) {
+            this._acsFipUuid = acsFipUuid;
+            return this;
+        }
+
+        public Builder egressDefaultPolicy(boolean egressDefaultPolicy) {
+            this._egressDefaultPolicy = egressDefaultPolicy;
+            return this;
+        }
+
+        @Override
+        public ImplementVspCommand build() {
+            return new ImplementVspCommand(_networkId, _networkDomainUuid, _networkUuid, _networkName, _vpcOrSubnetUuid, _isL2Network, _isL3Network, _isVpc, _isShared,
+                    _domainTemplateName, _isFirewallServiceSupported, _dnsServers, _ingressFirewallRules, _egressFirewallRules, _acsFipUuid, _egressDefaultPolicy);
+        }
+    }
+
+    @Override
+    public boolean executeInSequence() {
+        return false;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Implemented <code>equals</code> method for the <code>Command</code> super class, calling it from subclass.